server/src/api/board_sessions.go

```diff
@@ -101,3 +101,42 @@ func (s *Server) updateBoardSessions(w http.ResponseWriter, r *http.Request) {
 	render.Status(r, http.StatusOK)
 	render.Respond(w, r, sessions)
 }
+
+func (s *Server) updateReadyStates(w http.ResponseWriter, r *http.Request) {
```

Comment:
Hey @BenedictHomuth I just checked the code and saw that we already have this method implemented in `updateBoardSessions`. Why did you choose to add an additional endpoint, was there an error with the previous implementation?

Reply:
There is indeed an issue with the existing endpoint that can handle more than just the ready state.

This new endpoint sends out a bunch of `PARTICIPANT_UPDATED` event from the backend to the frontend for each participant. The existing endpoint sends *one* `PARTICIPANTS_UPDATED` event for all participants of a board. Note the missing `S` in the event of the new endpoint.

The problem with the `PARTICIPANTS_UPDATED` event is, that it is triggered by the [BoardSessionsObserver.AfterScanRow](https://github.com/inovex/scrumlr.io/blob/ea560aa1d9815a02e5118839a5c6434c2845019b/server/src/database/board_sessions_observer.go#L50) hook into the `bun` DB client. This hook is fired multiple times, not just once. It is triggered by [UpdateBoardSessions](https://github.com/inovex/scrumlr.io/blob/ea560aa1d9815a02e5118839a5c6434c2845019b/server/src/database/board_sessions.go#L163). And it is missing participants in its event payloads. Let's assume we have three participants on our board, P1, P2, P3. Then the `AfterScanRow` hook will be triggered thrice by `bun`. The first trigger sends an update with only `P1` in the participants list. Update 2 will have `P1` and `P2` in its payload. Update 3 finally sends out an update with all participants in the list.

This behavior is unexpected for the frontend that expects the `PARTICIPANTS_UPDATED` event to have a list of _all_ participants. The reducer for that event will look for the current session's user and if it can't find it, delete the `participants.self` entry from the global app state. A few components rely on that info to be present, though, basically throwing those users that are missing in the event's participant list from the board.

Reply:
There doesn't seem to be a way to `bun` to not trigger the hook multiple times. One solutino could be to switch the `PARTICIPANTS_UPDATED` event to a bunch of `PARTICIPANT_UPDATED` events. The number of events sent to the frontend by the backend is still the same.

Reply:
I tried to lookup a way to fix it with `bun` but it seems that there's no way. Instead I moved to call of the update into the `board_sessions.go` and left a comment there. Also I added a simple assertion in the tests that the update will be called only once. @BenedictHomuth and @migerh what do you think?
I tested it locally and everything seems to work fine.